diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +122,168 @@ class LineLength : Rule("line-length") {
+        } else leftInitOffset
+        val binList = mutableListOf<ASTNode>()
+        if (isProperty)
+            dfsForProperty(wrongNode, binList)
+        else
+            searchBinaryExpression(wrongNode, binList)
+        var binaryText = ""
+        binList.forEachIndexed { index, astNode ->
+            if (index == 0) {
+                binaryText += astNode.prevSibling { it.elementType == WHITE_SPACE}?.text ?: ""
+                binaryText += astNode.text
+                binaryText += astNode.nextSibling { it.elementType == WHITE_SPACE}?.text ?: ""
+                if (astNode.treeNext != null && astNode.treeNext.elementType == WHITE_SPACE)
+                    binaryText += astNode.treeNext.text
+            } else {
+                binaryText += astNode.prevSibling { it.elementType == WHITE_SPACE}?.text ?: ""
+                binaryText += astNode.treeParent.findChildByType(OPERATION_REFERENCE)!!.text
+                binaryText += astNode.text
+                binaryText += astNode.nextSibling { it.elementType == WHITE_SPACE}?.text ?: ""
+                if (leftOffset + binaryText.length > configuration.lineLength) {
+                    val commonParent = astNode.parent({ it in binList[index - 1].parents() })!!
+                    val nextNode = commonParent.findChildByType(OPERATION_REFERENCE)!!.treeNext
+                    if (!nextNode.text.contains("\n"))
+                        commonParent.appendNewlineMergingWhiteSpace(nextNode, nextNode)
+                    leftOffset = 0
+                    binaryText = astNode.text
+                    return
+                }
+            }
+        }
+    }
+
+    /**
+     * This method stored all the nodes that have BINARY_EXPRESSION or PREFIX_EXPRESSION element type.
+     * This method uses recursion to store binary node in the order in which they are located
+     * Also binList contains nodes with PREFIX_EXPRESSION element type ( !isFoo(), !isValid)
+     *@param node node in which to search
+     *@param binList mutable list of ASTNode to store nodes
+     */
+    private fun searchBinaryExpression(node: ASTNode, binList: MutableList<ASTNode>) {
+        if (node.hasChildOfType(BINARY_EXPRESSION) || node.hasChildOfType(PARENTHESIZED)) {
+            node.getChildren(null)
+                    .filter { it.elementType == BINARY_EXPRESSION || it.elementType == PARENTHESIZED }
+                    .forEach {
+                        searchBinaryExpression(it, binList)
+                    }
+        } else {
+            binList.add(node)
+            binList.add(node.treeParent.findChildByType(PREFIX_EXPRESSION) ?: return)
+        }
+    }
+
+    private fun dfsForProperty(node: ASTNode, binList: MutableList<ASTNode>) {
+        node.getChildren(null).forEach {
+            if (it.elementType in PROPERTY_LIST) {
+                if (it.elementType != REFERENCE_EXPRESSION || it.treeParent.elementType != SHORT_STRING_TEMPLATE_ENTRY)
```

Comment:
Please add comment explaining why these types are not included and what statements have these types.